### DISCUSSION TITLE
How much memory used by sway at boot?

### DISCUSSION THREAD

**Discussion by davidthewatson** [Manjaro-Sway/manjaro-sway#293] [Polls]
I have Manjaro sway installed on my ASUS UX 534 FTC with updates installed as of today 7/13/2022.

My system uses < 600 MB of memory with sway and a single foot terminal open to type "free". This seems way low. It doesn't break 2 GB when I have my development stack loaded with helix, python, chrome, foot, etc.

6-8 windows across 3 workspaces total 1.5 GB. This seems impossible but I don't think the kernel lies. I have no swap allocated since my Intel NVME breaks calamares on swap allocation.

**Comment by boredland** [comment]
this is hard for me to test, as I have quite some heave autostarts (mostly messengers, some electron..). But could try and kill all of them and report back.

**Reply by boredland** [reply]
on my x1 nano with some autostarts (slack, telegram, bitwarden, keybase):
```shell
$ free
               total        used        free      shared  buff/cache   available
Mem:           15722        2290       10802         541        2629       12574
Swap:          17297           0       17297
```

**Reply by boredland** [reply]
on my [x2100](https://www.xyte.ch/mods/x210-x2100/) without any autostarts that I knew of. so 1GB sounds reasonable:

```shell
$ free
               total        used        free      shared  buff/cache   available
Mem:           31981        1044       29966         324         969       30219
Swap:           9009           0        9009
```

**Reply by davidthewatson** [reply]
Interesting, those numbers are at least double what I'm seeing after accounting for autostarts. I've tested manjaro-sway on i7 booted clean at 512 MB consumed. Then, I tested my garuda sway on ryzen7 booted clean was nearly 1 GB, but not quite double the manjaro sway on i7. 

I'm going to install manjaro sway on a spare ryzen and see if i can duplicate the i7 low at 512 MB because it seems like an astounding holy grail for retro computing or those of us who favor geometry, speed, and function over beauty. 

It's a sweetspot I just have not experienced in many years. Tastes great, less filling.Thanks!

**Comment by davidthewatson** [comment]
Thanks!

Just curious.

FWIW, memory usage seemed extremely low and hard to follow with the new
updates this morning. Every time I took action, memory would act exactly
the opposite from what I'd expect, start a program, memory went down. Quit
a program, memory went up. I'm not sure what was going on, but it was
weird.
